lib/Analytics.js

```diff
@@ -36,6 +43,9 @@ export default class Analytics {
       }
     }
 
-    return fetch(`https://www.google-analytics.com/collect?${hit.toQueryString()}&${this.customDimensions.toQueryString()}&z=${Math.round(Math.random() * 1e8)}`, options);
+    if (this.userId !== null && this.userId !== undefined) {
```

Comment:
Seems like this could just be a falsy check, no? `if (this.userId) {` ? -- Also I'd probably prefer just `uid` as the variable name.